main.go

```diff
@@ -90,39 +92,40 @@ func main() {
 	submarinerInformerFactory := submarinerInformers.NewSharedInformerFactoryWithOptions(submarinerClient, time.Second*30,
 		submarinerInformers.WithNamespace(submSpec.Namespace))
 
-	start := func(context.Context) {
-		var localSubnets []string
+	var localSubnets []string
 
-		klog.Info("Creating the cable engine")
+	klog.Info("Creating the cable engine")
 
-		localCluster, err := util.GetLocalCluster(submSpec)
-		if err != nil {
-			klog.Fatalf("Error creating local cluster object from %#v: %v", submSpec, err)
-		}
+	localCluster, err := util.GetLocalCluster(submSpec)
+	if err != nil {
+		klog.Fatalf("Error creating local cluster object from %#v: %v", submSpec, err)
+	}
 
-		if len(submSpec.GlobalCidr) > 0 {
-			localSubnets = submSpec.GlobalCidr
-		} else {
-			localSubnets = append(submSpec.ServiceCidr, submSpec.ClusterCidr...)
-		}
+	if len(submSpec.GlobalCidr) > 0 {
+		localSubnets = submSpec.GlobalCidr
+	} else {
+		localSubnets = append(submSpec.ServiceCidr, submSpec.ClusterCidr...)
+	}
 
-		if len(submSpec.CableDriver) == 0 {
-			submSpec.CableDriver = cable.GetDefaultCableDriver()
-		}
-		submSpec.CableDriver = strings.ToLower(submSpec.CableDriver)
+	if len(submSpec.CableDriver) == 0 {
+		submSpec.CableDriver = cable.GetDefaultCableDriver()
+	}
+	submSpec.CableDriver = strings.ToLower(submSpec.CableDriver)
 
-		localEndpoint, err := util.GetLocalEndpoint(submSpec.ClusterID, submSpec.CableDriver, map[string]string{}, submSpec.NatEnabled,
-			localSubnets, util.GetLocalIP())
+	localEndpoint, err := util.GetLocalEndpoint(submSpec.ClusterID, submSpec.CableDriver, map[string]string{}, submSpec.NatEnabled,
+		localSubnets, util.GetLocalIP())
 
-		if err != nil {
-			klog.Fatalf("Error creating local endpoint object from %#v: %v", submSpec, err)
-		}
+	if err != nil {
+		klog.Fatalf("Error creating local endpoint object from %#v: %v", submSpec, err)
+	}
 
-		cableEngine, err := cableengine.NewEngine(localSubnets, localCluster, localEndpoint)
-		if err != nil {
-			klog.Fatalf("Fatal error occurred creating engine: %v", err)
-		}
+	cableEngine, err := cableengine.NewEngine(localSubnets, localCluster, localEndpoint, submarinerClient,
```

Comment:
We move this creation ahead in time, so ... the loop reconciling the gateway status will also
work for the PASSIVE gateway.

The driver initialization does not happen until "StartEngine" though.